tests/components/test_plant.py

```diff
@@ -34,30 +46,149 @@ def __init__(self, state=None):
-        # the JSON format than in hass
-        assert attrib[reading] == value
-
-
-@asyncio.coroutine
-def test_low_battery(hass):
-    """Test processing with low battery data and limit set."""
-    sensor = plant.Plant(hass, GOOD_CONFIG)
-    sensor.hass = hass
-    assert sensor.state_attributes['problem'] == 'none'
-    sensor.state_changed('sensor.mqtt_plant_battery',
-                         _MockState(45), _MockState(10))
-    assert sensor.state == 'problem'
-    assert sensor.state_attributes['problem'] == 'battery low'
+class TestPlant(unittest.TestCase):
+    """Tests for component "plant"."""
+
+    def setUp(self):
+        """Create test instance of home assistant."""
+        self.hass = get_test_home_assistant()
+        self.hass.start()
+
+    def tearDown(self):
+        """Stop everything that was started."""
+        self.hass.stop()
+
+    @asyncio.coroutine
+    def test_valid_data(self):
+        """Test processing valid data."""
+        sensor = plant.Plant('my plant', GOOD_CONFIG)
+        sensor.hass = self.hass
+        for reading, value in GOOD_DATA.items():
+            sensor.state_changed(
+                GOOD_CONFIG['sensors'][reading], None,
+                _MockState(value))
+        assert sensor.state == 'ok'
+        attrib = sensor.state_attributes
+        for reading, value in GOOD_DATA.items():
+            # battery level has a different name in
+            # the JSON format than in hass
+            assert attrib[reading] == value
+
+    @asyncio.coroutine
+    def test_low_battery(self):
+        """Test processing with low battery data and limit set."""
+        sensor = plant.Plant('other plant', GOOD_CONFIG)
+        sensor.hass = self.hass
+        assert sensor.state_attributes['problem'] == 'none'
+        sensor.state_changed('sensor.mqtt_plant_battery',
+                             _MockState(45), _MockState(10))
+        assert sensor.state == 'problem'
+        assert sensor.state_attributes['problem'] == 'battery low'
+
+    def test_update_states(self):
+        """Test updating the state of a sensor.
+
+        Make sure that plant processes this correctly.
+        """
+        plant_name = 'some_plant'
+        assert setup_component(self.hass, plant.DOMAIN, {
+            plant.DOMAIN: {
+                plant_name: GOOD_CONFIG
+            }
+        })
+        self.hass.states.set(MOISTURE_ENTITY, 5,
+                             {ATTR_UNIT_OF_MEASUREMENT: 'us/cm'})
+        self.hass.block_till_done()
+        state = self.hass.states.get('plant.'+plant_name)
+        self.assertEquals(STATE_PROBLEM, state.state)
+        self.assertEquals(5, state.attributes[plant.READING_MOISTURE])
+
+    def test_load_from_db(self):
+        """Test bootstrapping the brightness history from the database.
+
+        This test can should only be executed if the loading of the history
+        is enabled via plant.ENABLE_LOAD_HISTORY.
+        """
+        if plant.ENABLE_LOAD_HISTORY:
```

Comment:
It's better to use pytest skipif:
https://docs.pytest.org/en/latest/skipping.html#id1

Reply:
good point, changed the code